### PR TITLE
Adjust Gunboat's Scanner Loadout (to make getting scanned less of a joke)

### PR DESCRIPTION
**Pull request by Ferociousfeind**
**Bugfix:** I believe the "bugfix" section is most applicable. This PR addresses general notices that scanning has been "severely nerfed", by re-prioritizing the loadout of the one ship that actually regularly attempts to scan you: the humble Gunboat. It also serves to demonstrate what a ship that wants to scan should look like (and, again, if increasing scanner tonnage didn't break everyone's saves, I'd do that instead!)

## Fix Details
By making a small sacrifice to the battery capacity of the standard gunboat (it wasn't using it anyways), the captain can fit significantly more scanning oomph into the newly-vacated space. If they also focus on swarming the player, getting properly close for a good reading, even particularly large ships won't be able to "persist" or "resist" a gunboat's scanners for very long.

## Testing Done
I adjusted all the facets of the Gunboat's vanilla loadout to test what's functional to fly and what's on par with the pre-change loadout. Then, (once I realized the battery was size 3 and could be downgraded,) I flew the loadout I liked best through space, scanning random merchant, pirate, and navy ships in human space.

The loadout, with 9 cargo scanners (9t) and 7 outfit scanners (14t), has no trouble with any ships except the navy Carrier, which is has a large outfit space and so demands 5-6 seconds of being within scanning range to finish the scan. All other ships can be scanned in under 3 seconds if you're getting really close, or perhaps 5 seconds if you behave like a scanning NPC (only worried with keeping within maximum range to scan, boosting as the maximum range approaches.)

## Save File
N/A, any save could go visit human space to see how well spawned gunboats fare in attempting to scan the player's ships. Or, any save can install 9 cargo scanners and 7 outfit scanners (a total of 25t, and what the Gunboat is packing now) and try scanning some things. I think you'll find it's more usable than one of each (3t).

### CODE REVIEW

**Comment by Ferociousfeind** [comment]
Enlarging scanners was my preferred option, specifically to avoid having 1 scanner be not enough for a typical quick scan outing. But increasing scanner size would have untold consequences, especially in player save files. Everyone who installed a scanner previously would have (a chance to see) negative outfit space on their ship(s), and uninstslling the scanner would be a one-way ticket. And some nonzero number of vanilla outfit loadouts depend on scanners being specifically 1t, 2t, 4t.

(And the scaling of it all is ultimately arbitrary. There are magic numbers in the code to transform what I see as an appropriate amount of XYZ attribute into what I see as an appropriate XYZ-related utility. Why are cargo and outfit space of the target divided by 200? Completely arbitrary decision that "200t is pretty average outfit space, and only a fairly large cargo hold")

Perhaps this plays into it too- I'd like more chances to ask "how many" instead of "whether or not", when installing outfits on a ship. You only really get that with weapons, and only on ships with lots of gun ports or turret mounts. Everything else is more a "what size" than "how many" (if there are multiple sizes) and different sizes of outfits can be a bit tricky to track down.

(Sorry I'm sort of blowing up. I suppose I have a lot to say that I feel like hasn't been heard)

**Comment by Hecter94** [comment]
There are a number of variants of human/hai ships that are also equipped with Cargo/Outfit scanners. Do any of these also need to be updated?
"Bastion (Scanner)"
"Cruiser (Republic Intelligence)"
"Frigate (Republic Intelligence)"
"Quicksilver (Scanner)"
"Sparrow (Patrol)"
"Grasshopper (Surveillance)"
"Lightning Bug (Surveillance)"
"Shield Beetle (Police)"

**Comment by bene-dictator** [comment]
I don't think "Cruiser (Republic Intelligence)" is being used at the moment, and "Bastion (Scanner)" and "Quicksilver (Scanner)" are being used for smuggling missions, although I would really like for the Bastion one to be replaced by an Argosy. Not sure if any of the others are being used because I've never seen them...

**Comment by Hurleveur** [comment]
I think this could be pushing it a bit far? 3 of each could be sufficient?
I don't think we want the player getting instantly scanned without a chance to move away.

File: data/human/ships.txt [hunk 1]
```diff
@@ -1958,11 +1958,11 @@ ship "Gunboat"
...
 		
 		"RT-I Radiothermal"
-		"LP144a Battery Pack"
+		"LP072a Battery Pack"
 		"D14-RN Shield Generator"
 		"Small Radar Jammer"
-		"Cargo Scanner"
-		"Outfit Scanner"
+		"Cargo Scanner" 9
+		"Outfit Scanner" 7
```
Comment (Zitchas) [inline data/human/ships.txt:1965]:
```suggestion
		"Cargo Scanner" 8
		"Outfit Scanner" 3
```

This brings the cargo scan range from 900 down to 848 and outfit scan range from 1322 down to 866.

This will also increase the time to do a cargo scan by a small amount; and significantly increase the time to do an outfit scan.

**Review by Zitchas — CHANGES_REQUESTED** [review]
Honestly, scanning a Navy Carrier in 5-6 seconds and anything else in <3 is a touch on the "overboard" side. I'd expect such levels of performance from, say, a Navy Intelligence SigOps ship or other super-specialized/elite intel gathering ships. Regular gunboats should be at least double those time frames, potentially even triple. A regular gunboat scanning a Navy Carrier I'd expect to be more of a 20 to 30 seconds task. That's a *big* ship.

I'll also note that having such a common ship have such extremely short times basically pushes us back to the old "effectively insta-scan" environment, and removes any sort of reactionary capabilities from the player. They'd probably have to devote half their engine space to afterburners and keep their finger on the afterburner key, ready to slam it the instant they hear the scan noise in order to have even a hope of escaping it. I don't think this is particularly desireable. Making it a challenge to avoid/escape is good. Making it a "if you get within range of a gunboata you are toast" is not. 

Then there's also the range aspect. 7 outfit scanners gives 175 outfit scan power. The wiki states the range is 100 x the square root of the power. So that gives a range of 1322. Which is over 50% farther away than the longest range gun or turret in human space; and for a lot of people that will be well off screen. I know from my own testing that at many zoom levels, despite having a 1440p monitor, the closest edge of my screen is <800 away; and I find that the general combat/flight "sweet spot" seems to be to have my zoom set to a level that keeps the closest edge of my screen just slightly farther away than my longest range gun. Somewhere in the 800-1000 for me, typically.

So basically, this gunboat is going to scan me before I'm even likely to notice that it is specifically approaching me; and once it starts I'll have barely enough time to figure out which direction the gunboat is and figure out which direction I need to run in before it completes. The chance of getting *out* of its 1322 range in under 3 seconds is going to be hard. Very hard. 

---

Having the gunboat be better at scanning than it is now is 100% a thing I support. Having the stock gunboat gain performance that I'd equate to a specialized Navy Intelligence Sig Ops ship is not.

(having a *variant* gunboat with this many sanners is 100% OK, so long as it is a *rare* ship.

File: data/human/ships.txt (hunk 1 above)
Comment (Ferociousfeind) [reply data/human/ships.txt:1965]: Okay, you make good points. I, amazingly, hadn't considered the "ambushed while actually smuggling illicit cargo" perspective. I'll accept that, and cover the other scanning ships I've been made aware of.

(It should be noted, however, that pre-scan-change, the longest a scanner would take is 60 frames, one second. By comparison, 3 seconds is an improvement, even if only a modest one.)

Comment (Ferociousfeind) [reply data/human/ships.txt:1965]: I will also increase tactical scanner count by 1, so that the powers are 76, 75, and 64, which gives similar ranges to all three of them (848.5, 866.02, and 800 flat).

**Comment by quyykk** [comment]
We should probably address #7640 before tweaking variants

File: data/human/ships.txt (hunk 1 above)
Comment (Zitchas) [reply data/human/ships.txt:1965]: Sure, bumping up the tactical scanner by 1 seems reasonable.

**Comment by UnorderedSigh** [comment]
Alternative fix: https://github.com/endless-sky/endless-sky/pull/8288

**Comment by UnorderedSigh** [comment]
Okay, now #8288 also includes the aforementioned size formula adjustments. Illegal missions are about as hard as before, when small ships are used.

**Comment by UnorderedSigh** [comment]
After some quite thorough testing of #8288, I can tell you what is needed. Solo human scanners are practically incapable of dealing with large vessels. The Navy is not stupid; they know they need to scan large vessels. I suggest you add a purpose-built scanning Frigate with upgraded engines and an abundance of scanners. The "Frigate (Republic Intelligence)" used in Hai Reveal is almost what you need, but you need to replace the Catalytic Ramscoop and Brig with scanners. Of course, it also needs older weapons and shields.

If that Frigate shows up in maybe 1/3 of the Navy Surveillance fleets, players will learn to fear it. It will be *the reason* your illegal missions will unpredictably fail.

**Comment by Amazinite** [comment]
Closing because the state of the scanning mechanic itself is still up in the air (#9201), so it doesn't make much sense to leave PRs sit that mess with scanner stats or loadouts.
